test/src/unit-conversions.cpp

```diff
@@ -480,8 +494,7 @@ TEST_CASE("value conversion")
 #if defined(JSON_HAS_CPP_17)
         SECTION("std::string_view")
         {
-            std::string_view s = j;
-            CHECK(json(s) == j);
+            static_assert(!is_implicitly_convertible<decltype(j), std::string_view>::value);
```

Comment:
Why did you change the existing test?

Reply:
From the PR description:

The test was straight up failing and I implemented a new one. I came up with the test case from the [(missing) declaration of operator std::string_view](https://github.com/nlohmann/json/blob/develop/include/nlohmann/json.hpp#L2863) in `basic_json`.

Would you like me to elaborate on anything here?

Reply:
Could we add a check to only have the test if C++17 support is used?

Reply:
There's a ` #if defined(JSON_HAS_CPP_17)` in line 494 (line 480 in the old version).

Reply:
Ok. But why is the whole string_view test gone now?

Reply:
With the `JSON_HAS_STRING_VIEW` macro, is it possible to bring back the test

```cpp
SECTION("std::string_view")
{
    std::string_view s = j;
    CHECK(json(s) == j);
}
```

Reply:
I removed the template magic that would prevent the implicit conversion to `string_view` and adjusted the test accordingly. I'm still a bit worried that there was a good reason why the implicit conversion was disabled, though.

Reply:
I know that there has been discussion about removing all the implicit conversions.  I don't remember if it was completed or not.  This might have been part of that.

Reply:
No, the implicit conversions have not yet been removed. The test above worked so far, and if we merge this PR, it would be necessary that it still works as we otherwise would break existing implementations.